nodejs/awsx/ec2/vpcTopology.ts

```diff
@@ -291,13 +291,12 @@ ${lastAllocatedIpAddress} > ${lastVpcIpAddress}`);
                 }
 
                 const header = vpcIpv6CidrBlock.substr(0, colonColonIndex);
-                if (!header.endsWith("00")) {
-                    throw new pulumi.ResourceError(`Vpc ipv6 cidr block was not in an expected form: ${vpcIpv6CidrBlock}`, this.resource);
+                if (header.endsWith("00")) {
+                    const prefix = header.substr(0, header.length - 2);
+                    return prefix + index.toString().padStart(2, "0") + "::/64";
                 }
 
-                // trim off the 00, and then add 00, 01, 02, 03, etc.
-                const prefix = header.substr(0, header.length - 2);
-                return prefix + index.toString().padStart(2, "0") + "::/64";
+                return header + ":" + index.toString().padStart(2, "0") + "::/64";
```

Comment:
So, to clarify what's happening here - if it doesn't end in `00`, we're expecting it to have 1 less segment and so are just adding the colon as well as the incremental number?

Should we therefore be asserting the number of segments provided so it either has to end in `00` or has to have only 3 segments so we can add the last segment?